src/linker/Linker.Steps/LinkAttributesStep.cs

```diff
@@ -32,103 +32,150 @@ IEnumerable<CustomAttribute> ProcessAttributes (XPathNavigator nav, ICustomAttri
-							if (!Annotations.IsMarked (provider)) {
-								IEnumerable<Attribute> removeAttributeInstance = new List<Attribute> { new RemoveAttributeInstancesAttribute () };
-								Context.CustomAttributes.AddInternalAttributes (provider, removeAttributeInstance);
-							}
-							continue;
-						} else {
-							Context.LogWarning ($"Internal attribute 'RemoveAttributeInstances' can only be used on a type, but is being used on '{nav.Name}' '{provider}'", 2048, _xmlDocumentLocation);
-							continue;
-						}
-					} else {
-						Context.LogWarning ($"Unrecognized internal attribute '{internalAttribute}'", 2049, _xmlDocumentLocation);
-						continue;
-					}
+				if (!string.IsNullOrEmpty (internalAttribute)) {
+					ProcessInternalAttribute (provider, internalAttribute);
+					continue;
 				}
 
 				string attributeFullName = GetFullName (iterator.Current);
-				if (attributeFullName == String.Empty) {
+				if (attributeFullName == string.Empty) {
 					Context.LogWarning ($"Attribute element does not contain attribute 'fullname'", 2029, _xmlDocumentLocation);
 					continue;
 				}
-				string assemblyName = GetAttribute (iterator.Current, "assembly");
-				if (assemblyName == String.Empty)
-					attributeType = Context.GetType (attributeFullName);
-				else {
-					try {
-						assembly = GetAssembly (Context, AssemblyNameReference.Parse (assemblyName));
-					} catch (Exception) {
-						Context.LogWarning ($"Could not resolve assembly '{assemblyName}' in attribute '{attributeFullName}' specified in the '{_xmlDocumentLocation}'", 2030, _xmlDocumentLocation);
-						continue;
-					}
-					attributeType = assembly.FindType (attributeFullName);
-				}
-				if (attributeType == null) {
-					Context.LogWarning ($"Attribute type '{attributeFullName}' could not be found", 2031, _xmlDocumentLocation);
+
+				if (!GetAttributeType (iterator, attributeFullName, out TypeDefinition attributeType))
+					continue;
+
+				CustomAttribute customAttribute = GetCustomAttribute (iterator, attributeType);
+				if (customAttribute != null)
+					attributes.Add (customAttribute);
+			}
+
+			return attributes;
+		}
+
+		CustomAttribute GetCustomAttribute (XPathNodeIterator iterator, TypeDefinition attributeType)
+		{
+			string[] attributeArguments = (GetAttributeChildren (iterator.Current.SelectChildren ("argument", string.Empty))).ToArray ();
+			MethodDefinition constructor = attributeType.Methods.Where (method => method.IsInstanceConstructor ()).FirstOrDefault (c => c.Parameters.Count == attributeArguments.Length);
+			if (constructor == null) {
+				Context.LogWarning ($"Could not find a constructor for type '{attributeType}' that receives '{attributeArguments.Length}' arguments as parameter", 2022, _xmlDocumentLocation);
+				return null;
+			}
+
+			CustomAttribute customAttribute = new CustomAttribute (constructor);
+			var arguments = ProcessAttributeArguments (constructor, attributeArguments);
+			if (arguments == null)
+				return null;
```

Comment:
Here the custom attribute still exists just we don't want to execute the arguments code section, so you should return the customAttribute instead of null